cogs/thread_friends.py

```diff
@@ -0,0 +1,375 @@
+from utils import CommandChecks, TeXBotBaseCog
+from utils.error_capture_decorators import capture_guild_does_not_exist_error
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from collections.abc import Set as AbstractSet
+    from logging import Logger
+    from typing import Final
+
+    from utils import TeXBotApplicationContext, TeXBotAutocompleteContext
+
+
+ADD_COMMITTEE_TO_THREADS: "Final[bool]" = settings["ADD_COMMITTEE_TO_THREADS"]
+
+
+__all__: "Sequence[str]" = ("AddUsersToThreadsAndChannelsCog",)
+
+
+logger: "Final[Logger]" = logging.getLogger("TeX-Bot")
+
+
+class AddUsersToThreadsAndChannelsCog(TeXBotBaseCog):
+    """Cog for adding users to threads."""
+
+    @staticmethod
+    async def autocomplete_get_members(ctx: "TeXBotAutocompleteContext") -> "AbstractSet[discord.OptionChoice] | AbstractSet[str]":  # noqa: E501
+        """Autocomplete callable that generates the set of available selectable members."""
+        try:
+            main_guild: discord.Guild = ctx.bot.main_guild
+            guest_role: discord.Role = await ctx.bot.guest_role
+        except (GuildDoesNotExistError, GuestRoleDoesNotExistError):
+            return set()
+
+        members: set[discord.Member] = {
+            member
+            for member in main_guild.members
+            if not member.bot and guest_role not in member.roles
+        }
+
+        if not ctx.value or ctx.value.startswith("@"):
+            return {
+                discord.OptionChoice(name=f"@{member.name}", value=str(member.id))
+                for member in members
+            }
+
+        return {
+            discord.OptionChoice(name=member.name, value=str(member.id)) for member in members
+        }
+
+    @staticmethod
+    async def autocomplete_get_roles(ctx: "TeXBotAutocompleteContext") -> "AbstractSet[discord.OptionChoice] | AbstractSet[str]":  # noqa: E501
+        """Autocomplete callable that generates the set of available selectable roles."""
+        try:
+            main_guild: discord.Guild = ctx.bot.main_guild
+        except GuildDoesNotExistError:
+            return set()
+
+        if not ctx.value or ctx.value.startswith("@"):
+            return {
+                discord.OptionChoice(name=f"@{role.name}", value=str(role.id))
+                for role in main_guild.roles
+            }
+
+        return {
+            discord.OptionChoice(name=role.name, value=str(role.id))
+            for role in main_guild.roles
+        }
+
+    async def add_user_or_role_silently(self, user_or_role: discord.Member | discord.Role, thread: discord.Thread) -> None:  # noqa: E501
+        """Add a user or role to a thread without pinging them."""
+        message: discord.Message = await thread.send(
+            content=f"Adding {user_or_role!r} to thread...",
+            silent=True,
+        )
+        await message.edit(content=f"{user_or_role.mention}")
+        await message.delete(delay=1)
```

Comment:
Allow adding multiple users or roles to single thread
```suggestion
    async def add_user_or_role_silently(self, *users_and_roles: discord.Member | discord.Role, thread: discord.Thread) -> None:  # noqa: E501
        """Add a user or role to a thread without pinging them."""
        user_or_role: discord.Member | discord.Role
        for user_or_role in users_and_roles:
            message: discord.Message = await thread.send(
                content=f"Adding {user_or_role!r} to thread...",
                silent=True,
            )
            await message.edit(content=f"{user_or_role.mention}")
            await message.delete(delay=1)
```